python/cudf/cudf/utils/ioutils.py

```diff
@@ -1025,7 +1025,12 @@ def _is_local_filesystem(fs):
 
 
 def get_filepath_or_buffer(
-    path_or_data, compression, mode="rb", iotypes=(BytesIO), **kwargs
+    path_or_data,
+    compression,
+    mode="rb",
+    iotypes=(BytesIO),
+    allow_multiple_files=False,
```

Comment:
When would we not want to allow multiple files through here?

Reply:
We would want to not allow this for any reader that doesn't handle reading multiple files/buffers into a single dataframe. If `path_or_data` resolves to more than one file, we currently warn the user and read the first file only. Currently `read_parquet` is the only reader that supports reading multiple parquet files as a single dataframe.

Reply:
I'm wondering if we should throw instead of warn in these situations. That seems like a situation where if the user wants to read a single file they should be more explicit about it.

We should also probably throw in the individual readers as opposed to here.

Reply:
Ah, so would the recommended codepath something like? 
```
def reader(path):
    paths_or_buffers = get_filepath_or_buffer
    if len(paths_or_buffers) > 1:
        raise ...
```
One drawback here is that if the user originally gave a path that resolved to many large (remote) files, this method would potentially perform and expensive `f.read()` on all files before throwing. Open to alternate options

Reply:
Maybe create an alternative method of something like `_ensure_single_filepath_or_buffer`?